### PR TITLE
Jetpack Compose UI

### PR DESCRIPTION
**Pull request by safa-howaid**
Ready for review. Will need to refactor later to add finalized design.

### CODE REVIEW

File: android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+@Suppress("MagicNumber")
+@Composable
+fun SearchResultView(searchResult: SearchResult, onClick: () -> Unit) {
+    Card(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 20.dp)
+            .clickable(onClick = onClick)
+            .border(width = 0.5.dp, color = Color.LightGray),
```
Comment (dellisd) [inline android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt:106]:
```suggestion
            .border(width = 0.5.dp, color = MaterialTheme.color.onSurface.copy(alpha = ContentAlpha.disabled)),
```

File: android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    ) { Column(
+            modifier = Modifier
+                .padding(vertical = 5.dp)
+                .padding(horizontal = 10.dp),
+            horizontalAlignment = Alignment.Start,
+        ) {
+            Text(searchResult.name, style = typography.body1)
+            Text(searchResult.description, style = typography.body2, color = Color.DarkGray)
```
Comment (dellisd) [inline android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt:114]:
```suggestion
            CompositionLocalProvider(LocalContentAlpha provides ContentAlpha.medium) {
                Text(searchResult.description, style = typography.body2)
            }
```

File: android/src/main/java/com/cuhacking/atlas/ui/MapView.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+@Composable
+private fun MapViewContainer(map: MapView) {
+    AndroidView({ map }) { mapView ->
+        mapView.apply {
+            layoutParams = ViewGroup.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT,
+            )
+            getMapAsync { map ->
```
Comment (dellisd) [inline android/src/main/java/com/cuhacking/atlas/ui/MapView.kt:83]: Maybe not best for this PR, but we should try to find a way to control the `MapView` (composable) either via composable parameters, or by using a `State` object  of some kind. Similar to how other Jetpack Compose components work.

**Review by dellisd — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/cuhacking/atlas/ui/MapView.kt (hunk 3 above)
Comment (dellisd) [reply android/src/main/java/com/cuhacking/atlas/ui/MapView.kt:83]: For instance, the `ModalDrawer`: https://foso.github.io/Jetpack-Compose-Playground/material/modaldrawer/

File: android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt (hunk 1 above)
Comment (safa-howaid) [reply android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt:106]: ContentAlpha.disabled makes the results non-clickable, is that what we want?

Comment (dellisd) [reply android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt:106]: 🤔
That's not supposed to happen.

Comment (safa-howaid) [reply android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt:106]: actually nvm, might be from another change I made 😆

Comment (safa-howaid) [reply android/src/main/java/com/cuhacking/atlas/ui/SearchView.kt:106]: the update to beta 8 changed something, I'll look into it

**Review by dellisd — APPROVED** [review]

**Pull request merged**